study-group/Toast-Implementation-Studies.md

```diff
@@ -70,25 +70,164 @@ The `<paper-toast>` holds onto information about how it was dismissed the last t
+The final two parameters are used to offset the toast from its center of gravity, so this:
+
+```kotlin
+toast.setGravity(Gravity.TOP or Gravity.LEFT, 10, 10)
+```
+
+Will show the toast further down and further right.
+The toast can also be injected with a custom layout, which requires more complex Kotlin, explained [here](https://developer.android.com/guide/topics/ui/notifiers/toasts#CustomToastView).
+The `text`, `view`, `gravity`, and `duration` of the toast are all the elements the developer can control when working with this very straightforward component.
+
+
+### Notable Features & Details
+
+#### Gravity
+The library takes an interesting approach to positioning, by defining letting the user define the "center of gravity" of the element, along with offset values for the axes of gravity.
+
+#### Duration Constants
+The library comes with baked in constants for displaying the toast for long and short periods of time.
+
+### Takeaways
+- High opinion, low control.
+- The API for the `Toast` object provides getters and setters for each of `text`, `view`, `gravity`, `offset` and `duration`.
+- Beyond getters and setters, the only control the object allows is `.makeText()`, `.show()` and `.cancel()`.
+
+## [react-toastify](https://github.com/fkhadra/react-toastify)
+`react-toastify` is a React component library by author [@fkhadra](https://github.com/fkhadra) for rich, extensible notification toasts.
+
+### Design & Sample Code
+
+To use a `react-toastify` toast requires some simple JavaScript:
+
+```js
+class App extends Component {
+  notify = () => toast("Very simple toast!");
+
+  render(){
+    return (
+      <div>
+        <button onClick={this.notify}>Notify !</button>
+        <ToastContainer />
+      </div>
+    );
+  }
+}
+```
+
+`react-toastify` uses the explicit component pattern, by having the developer render a React `ToastContainer` component (the recommendation is to put this in the application root).
+This single component will render all toasts called from JavaScript, and can be configured with global options.
+However, the library allows the developer a lot of freedom, supporting multiple `ToastContainer`s as well as no `ToastContainer`, though one is recommended.
+When the `toast()` is called with a message and a props object, it is displayed in the toast container, with the toast props overriding the `ToastContainer` props, like so:
+
+```js
+class App extends Component {
+  tenSecondToast = () => toast("This toast will self destruct in 10...", { autoClose: 10000 });
+  normalToast = () => toast("This is a normal toast");
+
+  render(){
+    return (
+      <div>
+        <button onClick={this.tenSecondToast}>Close after 10 seconds</button>
+        <button onClick={this.normalToast}>Close after default amount of time</button>
+        <ToastContainer autoClose={5000} />
+      </div>
+    );
+  }
+}
+```
+
+The first button creates a toast that dismisses in 10 seconds, overriding the `ToastContainer` default.
```

Comment:
I'm not sure that this text is adding much value as the example also explains the outcome. Perhaps move anything extra here into the example?